server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -503,6 +503,20 @@
         Setting.Property.IndexScope
     );
 
+    /**
```

Comment:
Let's add another setting to enable

```
public static final Setting<Boolean> INDEX_MERGE_ON_FLUSH_ENABLED = Setting.timeSetting(
        "index.merge_on_flush.enabled",
        false,
        Property.IndexScope,
        Property.Dynamic
    );

```

Maybe we consider default true for 3.0?

Reply:
:+1:

Reply:
agree